solaris/utils/tile.py

```diff
@@ -600,8 +599,54 @@ def rasterize_gdf(gdf, src_shape, burn_value=1,
             ((geom, burn_value) for geom in gdf.geometry),
             out_shape=src_shape,
             transform=src_transform
-            )
+        )
     else:
         img = np.zeros(src_shape).astype(np.uint8)
 
     return img
+
+
+def vector_gdf_get_projection_unit(vector_file):
+    """Get the projection unit for a vector_file or gdf.
+
+    Arguments
+    ---------
+    vector_file : :py:class:`geopandas.GeoDataFrame` or geojson/shapefile
+        A vector file or gdf with georeferencing
+
+    Returns
+    -------
+    unit : String
+        The unit i.e. meters or degrees, of the projection
+    """
+    c = gpd.read_file(vector_file)
+    crs = c.crs
+    srs = osr.SpatialReference()
+    x = (crs['init']).split(":")[1]
+    srs.ImportFromEPSG(int(x))
+    WKT = srs.ExportToWkt()
+    unit = WKT.split("UNIT[")[1].split(",")[0]
+    return unit
+
+
+def raster_get_projection_unit(image):
+    """Get the projection unit for a vector_file.
+
+    Arguments
+    ---------
+    image : raster image, GeoTIFF or other format
+        A raster file with georeferencing
+
+    Returns
+    -------
+    unit : String
+        The unit i.e. meters or degrees, of the projection
+    """
+    c = rasterio.open(image)
```

Comment:
For consistency with the other functions that allow us to use either an already-loaded rasterio object or a path to a file, let's replace `rasterio.open` here with `_check_rasterio_im_load` from `sol.utils.core` (and update the docstring to reflect its flexibility).